microsoft/testsuites/vm_extensions/vmsnapshot_extension.py

```diff
@@ -81,16 +82,34 @@ def verify_vmsnapshot_extension(
         for count in range(10):
             try:
                 # create a restore point for the VM
-                restore_point = "rp_" + datetime.now().strftime("%Y-%m-%d-%H-%M-%S")
+                restore_point = (
+                    "rp_" + datetime.now().strftime("%Y-%m-%d-%H-%M-%S")
+                )
                 response = client.restore_points.begin_create(
                     resource_group_name=resource_group_name,
                     restore_point_collection_name=restore_point_collection,
                     restore_point_name=restore_point,
                     parameters={},
                 )
                 wait_operation(response, time_out=600)
-                log.info(f"restore point {restore_point} created")
-                break
+                # get the status of the restore point created
+                response = client.restore_points.get(
+                    resource_group_name=resource_group_name,
+                    restore_point_collection_name=restore_point_collection,
+                    restore_point_name=restore_point,
+                    expand=None)
+                if response.provisioning_state == "Succeeded":
+                    log.info(f"restore point {restore_point} created")
+                    consistency_mode = response.consistency_mode
+                    log.info(
+                        f"consistency mode is {consistency_mode}")
+                    if "File" in consistency_mode or "App" in consistency_mode:
+                        break
+                    else:
+                        raise ValueError(
+                            "Consistency mode is not File or App")
+                else:
+                    log.info(f"rp status is {response.provisioning_state}")
```

Comment:
if provisioning is not succeeded, check if we need to wait for some more time to check the completion status. The reason is that if it is in transitioning state, then subsequent create RP calls will fail too. We can fail the test with message that VM is in transitioning state and test can not proceed?